src/Components/Components/src/Rendering/RenderBatchBuilder.cs

```diff
@@ -15,6 +16,11 @@ namespace Microsoft.AspNetCore.Components.Rendering
     /// </summary>
     internal class RenderBatchBuilder : IDisposable
     {
+        // A value that, if changed, causes expiry of all ParameterView instances issued
+        // for this RenderBatchBuilder. This is to prevent invalid reads from arrays that
+        // may have been returned to the shared pool.
+        private int _parameterViewValidityStamp;
```

Comment:
Consider naming this `Version` or `Revision`. See https://github.com/aspnet/AspNetCore/blob/master/src/Http/Http.Features/src/FeatureCollection.cs#L27, https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/Infrastructure/ActionDescriptorCollection.cs#L39

Reply:
I was trying to avoid the implication that the values mean anything. The term "version" or "revision" implies that the value increases monotonically, and you can even do things like count the number of times it's changed.

In this case, it's more akin to identity's "security stamp" - it's just a value that can change, and all you know is whether or not it's now the same as some earlier snapshot.

I could factor this out into some other `struct` that hides the underlying value, but TBH that feels like overkill. It would be justified if this was public API, but it's all internal, so I think we can leave as-is and freely change later if any reason arises (which I doubt will happen).